tests/cloner_test.go

```diff
@@ -0,0 +1,133 @@
+		targetPod, err := f.FindPodByPrefix(common.CLONER_TARGET_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(strings.Compare(sourcePod.Spec.NodeName, targetPod.Spec.NodeName) == 0).To(BeTrue())
+
+		err = f.WaitTimeoutForPodStatus(sourcePod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitTimeoutForPodStatus(targetPod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		// Clone is completed, verify the content matches the source.
+		Expect(verifyTargetContent(f, f.Namespace, targetPvc)).To(BeTrue())
+		// Clean up PVC, the AfterEach will also clean it up, through the Namespace delete.
+		if targetPvc != nil {
+			err = f.DeletePVC(targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Cloner should clone data from PVC in name space to other PVC in different name space", func() {
+		targetNs, err := f.CreateNamespace(f.NsPrefix, map[string]string{
+			framework.NsPrefixLabel: f.NsPrefix,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		f.AddNamespaceToDelete(targetNs)
+		// Create targetPvc in new NS.
+		targetPvc, err := utils.CreatePVCFromDefinition(f.K8sClient, targetNs.Name, utils.NewPVCDefinition(
+			"target-pvc",
+			"1G",
+			map[string]string{controller.AnnCloneRequest: f.Namespace.Name + "/" + sourcePVCName},
+			nil))
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitForPersistentVolumeClaimPhase(f.K8sClient, targetNs.Name, v1.ClaimBound, targetPvc.Name)
+
+		sourcePod, err := f.FindPodByPrefix(common.CLONER_SOURCE_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+		targetPod, err := utils.FindPodByPrefix(f.K8sClient, targetNs.Name, common.CLONER_TARGET_PODNAME, common.CDI_LABEL_SELECTOR)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(strings.Compare(sourcePod.Spec.NodeName, targetPod.Spec.NodeName) == 0).To(BeTrue())
+
+		err = f.WaitTimeoutForPodStatus(sourcePod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitTimeoutForPodStatus(f.K8sClient, targetPod.Name, targetNs.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		// Clone is completed, verify the content matches the source.
+		Expect(verifyTargetContent(f, targetNs, targetPvc)).To(BeTrue())
+		// Clean up PVC, the AfterEach will also clean it up, through the Namespace delete.
+		if targetPvc != nil {
+			err = utils.DeletePVC(f.K8sClient, targetNs.Name, targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+	})
+
+})
+
+func verifyTargetContent(f *framework.Framework, namespace *v1.Namespace, pvc *v1.PersistentVolumeClaim) bool {
+	frameworkNs := f.Namespace
+	f.Namespace = namespace
+	executorPod, err := f.CreateExecutorPodWithPVC("verify-pvc-content", pvc)
+	Expect(err).ToNot(HaveOccurred())
+	err = f.WaitTimeoutForPodReady(executorPod.Name, utils.PodWaitForTime)
+	Expect(err).ToNot(HaveOccurred())
+	output := f.ExecShellInPod(executorPod.Name, "cat "+testFile)
+	f.DeletePod(executorPod)
```

Comment:
I'm still not convinced, FWIW, that deleting pods, pvc, etc (other than ns) should be handled by the  framewokr pkg. This feels to me more of a _utils_ pkg helper func.

Reply:
Its just delegating to the utils helper fuction with the namespace and client in the framework.